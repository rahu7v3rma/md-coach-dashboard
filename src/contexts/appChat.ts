import { createContext, useContext } from 'react';
import { StreamChat } from 'stream-chat';

const AppChatContext = createContext<{
    chatClient: StreamChat | null;
    setActiveChatChannel?: (channelId: string | null) => void;
}>({ chatClient: null, setActiveChatChannel: undefined });

const AppChatProvider = AppChatContext.Provider;

const useAppChat = () => {
    const { chatClient, setActiveChatChannel } = useContext(AppChatContext);
    return { chatClient, setActiveChatChannel };
};

export { AppChatProvider, useAppChat };
