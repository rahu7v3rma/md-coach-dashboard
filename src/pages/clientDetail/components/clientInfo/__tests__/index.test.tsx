import {
    RenderResult,
    fireEvent,
    render,
    screen
} from '@testing-library/react';
import { act } from 'react-dom/test-utils';

import ClientInfo from '..';
import { useAppChat } from 'src/contexts/appChat';

jest.mock('src/navigation');

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: jest.fn(() => mockNavigate)
}));

const mockChannelId = 'mockChannelId';
jest.mock('src/contexts/appChat', () => ({
    useAppChat: jest.fn()
}));
(useAppChat as jest.Mock).mockImplementation(() => ({
    chatClient: {
        userID: 1,
        queryChannels: jest.fn(() => Promise.resolve([{ cid: mockChannelId }]))
    }
}));

const props = {
    profileImg: 'profileImg',
    age: 'age',
    type: 'type'
};
global.URL.createObjectURL = jest.fn(() => props.profileImg);

const setup = async () => {
    let view!: RenderResult;
    await act(() => (view = render(<ClientInfo {...props} />)));
    const profileImage = view.container.querySelector(
        `img[src="${props.profileImg}"]`
    );
    await act(() => fireEvent.load(profileImage as HTMLElement));
    return view;
};

test('snapshot', async () => {
    const view = await setup();
    expect(view.asFragment()).toMatchSnapshot();
});

test('renders ClientInfo component with all props', async () => {
    const view = await setup();
    expect(view).toBeTruthy();
});

test('handleChatClick is not called when chatClient is null', async () => {
    (useAppChat as jest.Mock).mockImplementationOnce(() => ({
        chatClient: null
    }));
    const view = await setup();
    const chatButton = view.container.querySelector('button');
    await act(() => fireEvent.click(chatButton as HTMLElement));
    expect(mockNavigate).not.toBeCalled();
});

test('clicking the "Chat" button calls handleChatClick', async () => {
    const view = await setup();
    const chatButton = view.container.querySelector('button');
    await act(() => fireEvent.click(chatButton as HTMLElement));
    expect(mockNavigate).toBeCalledWith('/chat', {
        state: { channelId: mockChannelId }
    });
});

test('profile image is displayed when profileImg is provided', async () => {
    const view = await setup();
    const profileImage = view.container.querySelector(
        `img[src="${props.profileImg}"]`
    );
    expect(profileImage).toBeVisible();
});

test('profile image is not displayed when profileImg is null', async () => {
    let view!: RenderResult;
    await act(
        () => (view = render(<ClientInfo {...props} profileImg={null} />))
    );
    const profileImage = view.container.querySelector(
        `img[src="${props.profileImg}"]`
    );
    expect(profileImage).not.toBeInTheDocument();
    const defaultAvatar = screen.queryByAltText('avatar');
    expect(defaultAvatar).toBeInTheDocument();
});
