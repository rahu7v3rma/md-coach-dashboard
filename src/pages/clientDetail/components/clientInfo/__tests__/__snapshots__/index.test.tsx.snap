// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-cNNTdL ipNfQh"
  >
    <div
      class="sc-fFGjHI fiLKAe"
    >
      <div
        class="sc-dicizt fDkbuM sc-lpcdUm kedkeV"
      >
        <img
          alt=""
          src="profileImg"
        />
      </div>
    </div>
    <div
      class="sc-ehYhRg fALuif"
    >
      <p
        class="sc-jJTsDX hYccWd sc-fia-DRh kyMoJi"
      >
        name
      </p>
      <button
        class="sc-kPTPQs jWIHQp sc-iYsSXP kcemYU"
        height="40px"
        width="97px"
      >
        <img
          class="sc-eeGUxP cnJMqg"
          src="[object Object]"
        />
        <p
          class="sc-jJTsDX hYccWd sc-hYqwXO kxcvVT"
        >
          Chat
        </p>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
